src/EditorFeatures/CSharpTest/NavigateTo/NavigateToSearcherTests.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.NavigateTo;
+using Microsoft.CodeAnalysis.Navigation;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Moq;
+using Moq.Language.Flow;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.NavigateTo
+{
+    [UseExportProvider]
+    [Trait(Traits.Feature, Traits.Features.NavigateTo)]
+    public class NavigateToSearcherTests
```

Comment:
basic tests that validate the order of calls we get, and the relation between returning stale dta and being called in to get fresh data.